### PR TITLE
cudf and spark version info in artifacts

### PR DESCRIPTION
**Pull request by NvTimLiu**
1, cudf and spark version info in artifacts

    Add cudf and spark version info in rapids-plugin artifacts file:

    rapids4spark-version-info.properties

2, Only update Jenkins scripts, no source change. No unit tests needed.

3, No Github issue related to the PR.

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
In Jenkins pipeline log: 
20:20:55  cudf_version=cudf-0.15-20200814.090635-71-cuda10-1.jar
20:20:55  spark_version=spark-core_2.12-3.0.0.jar

In rapid-plugin jar file: rapids4spark-version-info.properties:
version=0.2.0-SNAPSHOT
user=
revision=536ae4b15d7a8ae35779867a6dcce253dd1883a6
branch=HEAD
date=2020-08-14T16:06:10Z
url=https://github.com/NvTimLiu/spark-rapids.git
CUDFVersion: cudf-0.15-20200814.090635-71-cuda10-1.jar
SPARKVersion: spark-core_2.12-3.0.0.jar

**Comment by NvTimLiu** [comment]
build

**Comment by jlowe** [comment]
> branch=HEAD

Something's going wrong there.  We want this to print the name of the branch being used during the build.  `HEAD` is not helpful.

File: build/build-info [hunk 1]
```diff
@@ -30,3 +30,7 @@ echo_build_properties() {
 }
 
 echo_build_properties $1
+
+# Parse cudf and spark files from local mvn repo
+${WORKSPACE}/jenkins/printJarVersion.sh "cudf_version" "${WORKSPACE}/.m2/ai/rapids/cudf/${CUDF_VER}" "cudf-${CUDF_VER}" "-${CUDA_CLASSIFIER}.jar"
+${WORKSPACE}/jenkins/printJarVersion.sh "spark_version" "${WORKSPACE}/.m2/org/apache/spark/spark-core_2.12/${SPARK_VER}" "spark-core_2.12-${SPARK_VER}" ".jar"
```
Comment (tgravescs) [inline build/build-info:36]: this doesn't handle anything with the shim layer versions of spark

**Comment by tgravescs** [comment]
what exactly are we using these for?

**Comment by NvTimLiu** [comment]
> > branch=HEAD
> 
> Something's going wrong there. We want this to print the name of the branch being used during the build. `HEAD` is not helpful.

Here I pasted `branch=HEAD`, because this is an MR build running on the pipeline `rapids_premerge-github`

On the nightly pipeline `rapids_nightly-0.2-github`, it will be `branch=branch-0.2`

**Comment by NvTimLiu** [comment]
> what exactly are we using these for?

@tgravescs  @jlowe Sorry for the confusion.

This PR is for the requirement from Sameer @sameerz.
He asked to put the cuDF & spark dependency version info into the rapids-plugin jar file.  I found there is a `rapids4spark-version-info.properties` file in the apids-plugin jar, so I just append cuDF & spark dependency version info in it.

Thanks!

**Comment by sameerz** [comment]
As @NvTimLiu mentioned, I requested we log the versions of Spark and cudf we are building the plugin with, so we have a record of the versions the plugin was built with.  Let us know if there is a better place to record the information along with the plugin.

**Comment by NvTimLiu** [comment]
> As @NvTimLiu mentioned, I requested we log the versions of Spark and cudf we are building the plugin with, so we have a record of the versions the plugin was built with. Let us know if there is a better place to record the information along with the plugin.


Ok, got it. Here are 2 ways below, which one do you prefer?

1) We tar version info into spark-rapid jars with another file name, e.g. dependency.properties, instead of `rapids4spark-version-info.properties`

2) Upload a file (with version info) out of the rapid jars, similar to: https://urm.nvidia.com/artifactory/sw-spark-maven-local/com/nvidia/rapids-4-spark_2.12/0.2.0-SNAPSHOT/rapids-4-spark_2.12-0.2.0-20200816.121112-88.pom

**Comment by tgravescs** [comment]
Personally I think it probably makes sense to be in a separate file, but we need to have the spark versions for all the shim layers as well, otherwise you only know what the core was built against.  For this you would have to look at the version in each of those pom files, at least that would be more reliable then passing it in and having to update it all the time.   For the databricks version we would want to make sure we get that separately and put it in that shim jar that we combine into the rest.

**Comment by NvTimLiu** [comment]
> Personally I think it probably makes sense to be in a separate file, but we need to have the spark versions for all the shim layers as well, otherwise you only know what the core was built against. For this you would have to look at the version in each of those pom files, at least that would be more reliable then passing it in and having to update it all the time. For the databricks version we would want to make sure we get that separately and put it in that shim jar that we combine into the rest.

@tgravescs, understand it, thanks for your detailed explanation.

Will add cudf & spark version info in a separate file. Besides, use the spark versions or all the shim layers instead of spark core versions

**Comment by NvTimLiu** [comment]
build

**Comment by tgravescs** [comment]
build

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
build

File: shims/spark300/pom.xml [hunk 2]
```diff
@@ -31,6 +31,46 @@
...
+            <plugin>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>dependency</id>
+                        <phase>generate-resources</phase>
+                        <configuration>
+                            <target>
+                                <mkdir dir="${project.build.directory}/extra-resources"/>
+                                <exec executable="bash" output="${project.build.directory}/extra-resources/${spark300.version}-info.properties">
```
Comment (tgravescs) [inline shims/spark300/pom.xml:47]: it might be good to prefix with spark for the file name spark-300-info.properties

**Comment by NvTimLiu** [comment]
@jlowe @tgravescs @sameerz  Could you please help to check if the PR meet your requirement? Thanks!

**Comment by tgravescs** [comment]
@NvTimLiu  I left you a comment, I am waiting for you to address it

**Comment by NvTimLiu** [comment]
> so the dist package will automatically pull in all the properties file when it creates the assembly jar, correct?

Yes, correct.

File: shims/spark300/pom.xml (hunk 2 above)
Comment (NvTimLiu) [reply shims/spark300/pom.xml]: Agree, added prefix with `spark` for the version files

**Comment by NvTimLiu** [comment]
build

**Pull request merged**